src/generateDiff.js

```diff
@@ -0,0 +1,41 @@
+const diff = require('diff');
+
+/**
+ * Generates a diff between two multi-line strings. The resulting diff shows
+ * any changes using '-' and '+' to indicate the "old" and "new" version
+ * respectively, and includes 2 lines of unchanged content around each changed
+ * section where possible.
+ * @param {string} before - The string representing the base for the comparison.
+ * @param {string} after - The string representing the changes being compared.
+ * @returns {string} The genereated text diff
+ */
+function generateDiff(before, after) {
+  const changes = diff.diffLines(before, after);
+  const diffLines = [];
+  const preceedingContext = [];
+  for (const { added, removed, value } of changes) {
+    const lines = value.split('\n');
+    // remove trailing newline
+    lines.pop();
```

Comment:
I didn't know actually. I meant to come back to this and forgot. This is a mistake I believe.

Because `diffLines` is meant to operate on lines, I had wrongly assumed that the newlines between each changeset would be removed. That is not the case though, so when I'm joining with `\n` below I'm introducing a _new_ newline that I've removed here to compensate.

I'll update this to just preserve the newline in the text.

Reply:
Well, this ended up being much more complicated than anticipated! I tried and failed to fix this many times, namely because of the ambiguity introduced when files don't end in newlines.

I ended up writing comprehensive tests for `generateDiff`, and rewriting most of the diff logic. Result here: b1375e802173782feb4c6e2dccf98d3f081a2dbf